doc/design/networking/Investigation-on-NSX-integration.md

```diff
@@ -0,0 +1,254 @@
+
+### Create VCH 
+Specifying 'bridge-network' as the port group 'vxw-dvs-55-virtualwire-1-sid-5000-logical-switch-1'.
+  * VIC creates VCH
+  * NSX manager creates a security group `<vch name>-SG-VCH` with dynamic membership: NIC (of VCH) 
+  * NSX manager creates these security policies shown in the following picture
+  * NSX manager applies security policy `<vch name>-SP-0` on security group `<vch name>-SG-VCH`
+
+  ![Security policies](pics/security-policies.png)
+
+### docker run --name=containerVM1 <image name>
+  * VIC create containerVM1
+  * NSX manager creates security group `<vch name>-SG-bridge` with dynamic membership: NIC-1 (of containerVM1)
+  * NSX manager applies all the defined policy groups to this security group
+
+### docker network create network-a
+  * VIC creates network scopes for this container bridge network
+  * NSX manager creates a security group `<vch name>-SG-network-a` 
+  * NSX manager applies all the defined policy groups to this security group  
+  (creating network-b is similar are similar to this)
+
+### docker run -net=network-a --name=containerVM3 <image name>
+  * VIC creates containerVM3
+  * NSX manager update security group `<vch name>-SG-network-a` with dynamic mebership: NIC-3 (of containerVM3)
+  (creating other containerVMs are similar to this)
+
+### Summary of security groups and security policies of this example
+We associate vNICs to security groups. So the memberships of security groups look like this :
+  * `<vch name>-SG-VCH`  dynamic membership: NIC (of VCH) 
+  * `<vch name>-SG-bridge`    dynamic membership: NIC-1 (of containerVM1), NIC-2 (of containerVM2)
+  * `<vch name>-SG-network-a`    dynamic membership: NIC-3 (of containerVM3), NIC-4 (of containerVM4), NIC-52 (of containerVM5)
+  * `<vch name>-SG-network-b`    dynamic membership: NIC-5 (of containerVM5), NIC-6 (of containerVM6)
+
+We apply the following security policies to our security groups:
+  * `<vch name>-SP-0` is applied to all the security groups
+  * `<vch name>-SP-1` is applied to security groups `<vch name>-SG-bridge`, `<vch name>-SG-network-a` and `<vch name>-SG-network-b` 
+  * `<vch name>-SP-2` is applied to security groups `<vch name>-SG-bridge`, `<vch name>-SG-network-a` and `<vch name>-SG-network-b`
+  * `<vch name>-SP-3` is applied to security groups `<vch name>-SG-bridge`, `<vch name>-SG-network-a` and `<vch name>-SG-network-b`
+
+The result we get :
+  * all the networks are isolated to each other
+  * containerVMs in the network can connect to their gateway at VCH
+
+## NSX Manager API:
+![Code flow to use NSX Manager API](pics/with-nsx.png)
+
+### API endpoint
+https://{nsxmanager's hostname or IP}/api 
+
+### API calls
+(details in http://pubs.vmware.com/nsx-63/topic/com.vmware.ICbase/PDF/nsx_63_api.pdf , note that many NSX API methods reference vCenter object IDs in URI parameters, query parameters, request bodies, and response bodies, also it takes XML as request body (sad) )
```

Comment:
it is a REST API... I think the plan is to add code in govmomi. Then make calls to govmomi from port layer. I'm not familiar with wsdl..what is it? and do you think at least I can use it to handle the xml part?